assets/src/domain/eventEditor/ui/datetimes/dateForm/useDateFormConfig.ts

```diff
@@ -10,147 +10,161 @@ import { EntityId } from '@dataServices/types';
+import { setDateToToday, setTimeToZeroHour, setDefaultTime } from '@appServices/utilities/date';
+import { useMemo, useCallback } from 'react';
 
 type DateFormConfig = EspressoFormProps<DateFormShape>;
 
 const FIELD_NAMES: Array<keyof Datetime> = ['id', 'name', 'description', 'capacity', 'isTrashed'];
 
 const useDateFormConfig = (id: EntityId, config?: EspressoFormProps): DateFormConfig => {
-	const { startDate: start, endDate: end, ...restProps } = useDatetimeItem({ id }) || {};
+	const datetime = useDatetimeItem({ id });
 
 	const { siteTimeToUtc, utcToSiteTime } = useTimeZoneTime();
 
-	const startDate = start ? utcToSiteTime(parseISO(start)) : PLUS_ONE_MONTH;
-	const endDate = end ? utcToSiteTime(parseISO(end)) : PLUS_TWO_MONTHS;
+	const startDate = useMemoStringify(
+		datetime?.startDate ? utcToSiteTime(parseISO(datetime?.startDate)) : setDefaultTime(PLUS_ONE_MONTH, 'start')
+	);
+	const endDate = useMemoStringify(
+		datetime?.endDate ? utcToSiteTime(parseISO(datetime?.endDate)) : setDefaultTime(PLUS_ONE_MONTH, 'end')
+	);
 
 	const { onSubmit } = config;
 
-	const onSubmitFrom: DateFormConfig['onSubmit'] = ({ dateTime, ...rest }, form, ...restParams) => {
-		// convert "dateTime" object to proper UTC "startDate" and "endDate"
-		const { startDate, endDate } = processDateAndTime(dateTime, siteTimeToUtc);
+	const onSubmitFrom: DateFormConfig['onSubmit'] = useCallback(
+		({ dateTime, ...rest }, form, ...restParams) => {
+			// convert "dateTime" object to proper UTC "startDate" and "endDate"
+			const { startDate, endDate } = processDateAndTime(dateTime, siteTimeToUtc);
 
-		const values = { ...rest, startDate, endDate };
+			const values = { ...rest, startDate, endDate };
 
-		return onSubmit(values, form, ...restParams);
-	};
-
-	const initialValues: DateFormShape = {
-		...pick<Partial<Datetime>, keyof Datetime>(FIELD_NAMES, restProps),
-		dateTime: {
-			/**
-			 * for validations, we need to make
-			 * - time same for dates
-			 * - date same for times
-			 * ¯\_(ツ)_/¯
-			 */
-			startDate: setTimeToZeroHour(startDate),
-			startTime: setDateToToday(startDate),
-			endDate: setTimeToZeroHour(endDate),
-			endTime: setDateToToday(endDate),
+			return onSubmit(values, form, ...restParams);
 		},
-	};
+		[onSubmit, siteTimeToUtc]
+	);
+
+	const initialValues: DateFormShape = useMemo(
+		() => ({
+			...pick<Partial<Datetime>, keyof Datetime>(FIELD_NAMES, datetime || {}),
+			dateTime: {
+				/**
+				 * for validations, we need to make
+				 * - time same for dates
+				 * - date same for times
+				 * ¯\_(ツ)_/¯
+				 */
+				startDate: setTimeToZeroHour(startDate),
+				startTime: setDateToToday(startDate),
+				endDate: setTimeToZeroHour(endDate),
+				endTime: setDateToToday(endDate),
+			},
+		}),
+		[datetime, endDate, startDate]
+	);
 
-	const adjacentFormItemProps = {
+	const adjacentFormItemProps = useMemoStringify({
 		className: 'ee-form-item-pair',
-	};
+	});
```

Comment:
why are we memoizing this?
because a new object is created every time?

Reply:
yes